src/gt4py/next/program_processors/runners/dace_fieldview/transformations/gpu_utils.py

```diff
@@ -0,0 +1,374 @@
+        allow_none=False,
+        default=(32, 1, 1),
+        setter=_gpu_block_parser,
+        getter=_gpu_block_getter,
+        desc="Size of the block size a GPU Map should have.",
+    )
+
+    launch_bounds = properties.Property(
+        dtype=str,
+        allow_none=True,
+        default=None,
+        desc="Set the launch bound property of the map.",
+    )
+
+    map_entry = transformation.transformation.PatternNode(nodes.MapEntry)
+
+    def __init__(
+        self,
+        block_size: Sequence[int | str] | str | None = None,
+        launch_bounds: int | str | None = None,
+        launch_factor: int | None = None,
+    ) -> None:
+        super().__init__()
+        if block_size is not None:
+            self.block_size = block_size
+
+        if launch_factor is not None:
+            assert launch_bounds is None
+            self.launch_bounds = str(
+                int(launch_factor) * self.block_size[0] * self.block_size[1] * self.block_size[2]
+            )
+        elif launch_bounds is None:
+            self.launch_bounds = None
+        elif isinstance(launch_bounds, (str, int)):
+            self.launch_bounds = str(launch_bounds)
+        else:
+            raise TypeError(
+                f"Does not know how to parse '{launch_bounds}' as 'launch_bounds' argument."
+            )
+
+    @classmethod
+    def expressions(cls) -> Any:
+        return [dace.sdfg.utils.node_path_graph(cls.map_entry)]
+
+    def can_be_applied(
+        self,
+        graph: Union[SDFGState, SDFG],
+        expr_index: int,
+        sdfg: dace.SDFG,
+        permissive: bool = False,
+    ) -> bool:
+        """Test if the block size can be set.
+
+        The function tests:
+        - If the block size of the map is already set.
+        - If the map is at global scope.
+        - If if the schedule of the map is correct.
+        """
+
+        scope = graph.scope_dict()
+        if scope[self.map_entry] is not None:
+            return False
+        if self.map_entry.map.schedule not in dace.dtypes.GPU_SCHEDULES:
+            return False
+        if self.map_entry.map.gpu_block_size is not None:
+            return False
+        return True
+
+    def apply(
+        self,
+        graph: Union[SDFGState, SDFG],
+        sdfg: SDFG,
+    ) -> None:
+        """Modify the map as requested."""
+        self.map_entry.map.gpu_block_size = self.block_size
+        if self.launch_bounds is not None:  # Note empty string has a meaning in DaCe
+            self.map_entry.map.gpu_launch_bounds = self.launch_bounds
+
+
+@properties.make_properties
+class SerialMapPromoterGPU(transformation.SingleStateTransformation):
+    """Serial Map promoter for empty Maps in case of trivial Maps.
+
+    In CPU mode a Tasklet can be outside of a map, however, this is not
+    possible in GPU mode. For this reason DaCe wraps such Tasklets in a
+    trivial Map.
+    This transformation will look for such Maps and promote them, such
```

Comment:
I see the benefit of this transformation. We are moving program code (I mean computation) to each CUDA thread. I do not know the kind of computation well enough to say whether it could be interesting to instead compute this data on CPU and access it from GPU (implies a device to host read, which eventually is not too expensive on gh200). Removing computation from the GPU is a way to reduce register utilisation, I guess.

Reply:
I see what you are mean.
But as far as I understand GT4Py having trivial Maps is most likely an extreme edge case.
Currently, Tasklets outside of Maps are only these "literal Tasklets", by moving them inside teh same kernel nvcc will detect them as literal/constant.

if you put them in another kernel/compilation unit, nvcc will never know that it is a literal so it will always result in a memory load, which is slower than a literal (which could blow up binary size) or a register.